### PR TITLE
Added Issue.as_pull_request() and PullReqest.as_issue()

### PR DESCRIPTION
**Pull request by tmshn**
As reported in #572, there's no straightforward way to convert from `github.Issue.Issue` to `github.PullRequest.PullRequest` (when it is actually a pull request), and vice versa.

A workaround is introduced there, but that is ugly and may be confusing for readers.

```python
>>> issue.repository.get_pull(issue.number)
```

To fix this, I added `Issue.as_pull_request()` and `PullReqest.as_issue()` methods to do this.
Internally it calls same api as the workaround does, but way clearer.

```python
>>> issue.as_pull_request()
```

### CODE REVIEW

File: github/Issue.py [hunk 1]
```diff
@@ -230,6 +230,17 @@ def user(self):
...
         return self._user.value
 
+    def as_pull_request(self):
+        """
+        :calls: `GET /repos/:owner/:repo/pulls/:number <http://developer.github.com/v3/pulls>`_
+        :rtype: :class:`github.PullRequest.PullRequest`
+        """
+        headers, data = self._requester.requestJsonAndCheck(
+            "GET",
+            "/pulls/".join(self.url.rsplit("/issues/", 1))
```
Comment (tmshn) [inline github/Issue.py]: This is the hack to replace only last occurrence of the pattern; just `self.url.replace("/issues/", "/pulls/")` will replace all, which might be user name or repo name.
https://stackoverflow.com/questions/2556108/rreplace-how-to-replace-the-last-occurrence-of-an-expression-in-a-string

**Comment by sfdye** [comment]
@tmshn Thanks very much for your PR. However now as #572 is close, I don't feel the necessity for this PR anymore. I am afraid I will close it for now.

File: github/Issue.py (hunk 1 above)
Comment (sfdye) [reply github/Issue.py]: Could there be more than one occurrence? If not, why not just use `str.replace()`

Comment (tmshn) [reply github/Issue.py]: I don't know much about the Github specification, but I cannot guarantee that following situation never happen:

- User/organization name or repository name is `"issues"` or `"pulls"`
- Hostname of Github Enterprise is configured to something like `"mycompany.com/issues/ghe"`

Of course it's ridiculous idea to do that, but if it's POSSIBLE, we should care that.

If we can say this never happens, I will use `str.replace()`.

Comment (sfdye) [reply github/Issue.py]: You are absolutely right. Thanks for being so prudent.

**Pull request merged**